src/tools/rbd/action/MirrorImage.cc

```diff
@@ -282,6 +283,20 @@ int execute_status(const po::variables_map &vm,
     return r;
   }
 
+  std::string instance_id;
+  MirrorDaemonServiceInfo daemon_service_info(io_ctx);
+
+  if (status.up) {
+    r = image.mirror_image_get_instance_id(&instance_id);
+    if (r < 0 && r != -ENOENT) {
```

Comment:
Nit: one last thing, this should probably filter our ```-EOPNOTSUPP``` to avoid the case where older OSDs are being used. If the OSDs' don't support this new cls method, don't include the status.

Reply:
@dillaman  I am not sure about this. With the current code it will print the error (warning) but still will output the status. So the user will be informed that the information is not full and will know the reason of this. It differs from `-ENOENT` case, because it means that information was not stored (does not exist), while for `-EOPNOTSUPP` it means that it may be their but we can't obtain it.

But I don't have strong opinion here and will change it if you still think it is better to suppress the error message in this case.

Reply:
If a special-case were to be added for ```-EOPNOTSUPP```, a more friendly message could be provided like ```rbd: newer release of Ceph OSDs required to map image to rbd-mirror daemon instance``` (or similar)